pkg/terminal/auth.go

```diff
@@ -1,33 +1,22 @@
 package terminal
 
 import (
-	"context"
-
-	authv1 "k8s.io/api/authorization/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// checkUserPermissions checks if the terminal proxy is supported for a given user.
-// Returns true if we're willing to proxy the user's token, false otherwise. We don't
-// want to proxy highly privileged tokens to avoid privilege escalation issues.
-func (p *Proxy) checkUserPermissions(token string) (bool, error) {
-	client, err := p.createTypedClient(token)
+// isClusterAdmin checks if the user belongs to the cluster-admin group
+func (p *Proxy) isClusterAdmin(userInfo *unstructured.Unstructured, namespace string) (bool, error) {
+	groups, isFound, err := unstructured.NestedStringSlice(userInfo.UnstructuredContent(), "groups")
 	if err != nil {
 		return false, err
 	}
-
-	sar := &authv1.SelfSubjectAccessReview{
-		Spec: authv1.SelfSubjectAccessReviewSpec{
-			ResourceAttributes: &authv1.ResourceAttributes{
-				Namespace: "openshift-operators",
-				Verb:      "create",
-				Resource:  "pods",
-			},
-		},
+	if !isFound {
+		return false, nil
 	}
-	res, err := client.AuthorizationV1().SelfSubjectAccessReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
-	if err != nil || res == nil {
-		return false, err
+	for _, group := range groups {
+		if group == "system:cluster-admins" {
```

Comment:
Again, I don't think this is a reliable way to determine if the user is a cluster admin.

Reply:
@spadgett is there a consistent way or better way to do this?